### PR TITLE
feat: Sentry FastBoot Support

### LINKED ISSUE #4092: Add sentry fastboot integration
Currently only client errors are reported through sentry, whereas production runs in fastboot mode and it is the major source of code ran and all events are dropped from there

### PR DESCRIPTION
**Pull request by kushthedude**
Fixes #4092

### CODE REVIEW

File: scripts/fastboot-server.js [hunk 1]
```diff
@@ -10,17 +10,25 @@ const fastbootHost = process.env.FASTBOOT_HOST || '0.0.0.0';
...
   gzip            : enableGzip === 'true',
   host            : fastbootHost,
   port            : parseInt(fastbootPort),
   chunkedResponse : enableChunkedResponse === 'true',
   workerCount     : parseInt(fastbootWorkers)
-});
+};
+
+if (process.env.SENTRY_DSN) {
+  Sentry = require('@sentry/node');
```
Comment (niranjan94) [inline scripts/fastboot-server.js]: Codacy Issue found: ['Sentry' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 22
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987&bid=16530086)

File: scripts/fastboot-server.js [hunk 2]
```diff
@@ -10,17 +10,25 @@ const fastbootHost = process.env.FASTBOOT_HOST || '0.0.0.0';
...
+  Sentry.init({ dsn: process.env.SENTRY_DSN });
```
Comment (niranjan94) [inline scripts/fastboot-server.js]: Codacy Issue found: ['Sentry' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

File: app/services/error-handler.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      const originalUrl = (req.originalUrl || req.url || req.path);
+      const absoluteUrl = `${protocol}://${host}${originalUrl}`;
+      // const userScope = this.userScope; // If you have an user scope you can get from service property here
+
+      FastBootRaven.withScope(function(scope) {
+        scope.setTag('url', absoluteUrl);
+        scope.setExtra('query_params', req.query_params);
+        // scope.setUser(userScope);
+
+        FastBootRaven.captureException(error);
```
Comment (niranjan94) [inline app/services/error-handler.js]: Codacy Issue found: ['FastBootRaven' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

File: app/services/error-handler.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { inject as service } from '@ember/service';
+
+export default class errorHandlerService extends Service {
+  @service('fastboot') fastboot;
+
+  handleServerSideError(error) {
+    if (!this.fastboot.isFastBoot) {return}
+
+    if (typeof FastBootRaven !== 'undefined') {
+      const req = fastboot.request;
```
Comment (niranjan94) [inline app/services/error-handler.js]: Codacy Issue found: ['fastboot' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

File: app/services/error-handler.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
```
Comment (niranjan94) [inline app/services/error-handler.js]: Codacy Issue found: ['@ember/service' import is duplicated.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

File: app/services/error-handler.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    if (typeof FastBootRaven !== 'undefined') {
+      const req = fastboot.request;
+      const { protocol, headers } = req;
+      const host = req.hostname || req.host || headers.host || '<no host>';
+      const originalUrl = (req.originalUrl || req.url || req.path);
+      const absoluteUrl = `${protocol}://${host}${originalUrl}`;
+      // const userScope = this.userScope; // If you have an user scope you can get from service property here
+
+      FastBootRaven.withScope(function(scope) {
```
Comment (niranjan94) [inline app/services/error-handler.js]: Codacy Issue found: ['FastBootRaven' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

File: scripts/fastboot-server.js [hunk 7]
```diff
@@ -10,17 +10,25 @@ const fastbootHost = process.env.FASTBOOT_HOST || '0.0.0.0';
...
+  serverOptions.sandboxGlobals = { FastBootRaven: Sentry };
```
Comment (niranjan94) [inline scripts/fastboot-server.js]: Codacy Issue found: ['Sentry' is not defined.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5011987)

**Comment by kushthedude** [comment]
@iamareebjamal 
- There were lint issue with sentry.ts as it didn't follow the indents guidelines of the project, hence fixing the minor change with this PR.
- There is ember-cli-sentry which I will add in the next commit for the proper error tracking, also there is no fastboot support for same and hence this is a workaround for error catching in fastboot environment as explained here https://github.com/ember-cli-sentry/ember-cli-sentry/issues/77#issuecomment-547066430

**Comment by iamareebjamal** [comment]
Apart from changes to fix lint issues, the only changes required to make it work with fastboot are to be done in sentry.ts

Just check if it is fastboot or not and require sentry/node instead of sentry/browser

No need to do anything in fastboot app as it'll initialize sentry twice

**Comment by iamareebjamal** [comment]
ember-cli-sentry is not needed. Need to move away from ember addons and middlemen as much as possible

**Pull request closed without merging**